src/ts/components/ecomodetogglebutton.ts

```diff
@@ -0,0 +1,86 @@
+import { ToggleButton, ToggleButtonConfig } from './togglebutton';
+import { UIInstanceManager } from '../uimanager';
+import { DynamicAdaptationConfig, PlayerAPI, VideoQualityChangedEvent } from 'bitmovin-player';
+import { i18n } from '../localization/i18n';
+
+export class EcoModeToggleButton extends ToggleButton<ToggleButtonConfig> {
+  private adaptationConfig: DynamicAdaptationConfig;
+
+  constructor(config: ToggleButtonConfig = {}) {
+    super(config);
+
+    const defaultConfig: ToggleButtonConfig = {
+      text: i18n.getLocalizer('ecoMode'),
+      cssClass: 'ui-ecomodetogglebutton',
+      onClass: 'on',
+      offClass: 'off',
+      ariaLabel: i18n.getLocalizer('ecoMode'),
+    };
+
+    this.config = this.mergeConfig(config, defaultConfig, this.config);
+  }
+
+  configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
+    super.configure(player, uimanager);
+
+    if (this.areAdaptationApisAvailable(player)) {
+      this.onClick.subscribe(() => {
+        this.toggle();
+      });
+
+      this.onToggleOn.subscribe(() => {
+        this.enableEcoMode(player);
+        player.setVideoQuality('auto');
+      });
+
+      this.onToggleOff.subscribe(() => {
+        this.disableEcoMode(player);
+      });
+
+      player.on(player.exports.PlayerEvent.VideoQualityChanged, (quality: VideoQualityChangedEvent) => {
+        if (quality.targetQuality.height !== null) {
+          this.off();
+          this.disableEcoMode(player);
+        }
+      });
+    } else {
+      super.disable();
+    }
+
+  }
+
+  private areAdaptationApisAvailable(player: PlayerAPI): boolean {
+    const isGetConfigAvailable = Boolean(player.adaptation.getConfig && typeof player.adaptation.getConfig === 'function');
+    const isSetConfigAvailable = Boolean(player.adaptation.setConfig && typeof player.adaptation.setConfig === 'function');
+
+    return Boolean(player.adaptation && isGetConfigAvailable && isSetConfigAvailable);
+  }
+
+  enableEcoMode(player: PlayerAPI): void {
+    this.adaptationConfig = player.adaptation.getConfig();
+    let codec = player.getAvailableVideoQualities()[0].codec;

```

Comment:
NIT: should be const

Reply:
[Use const instead of let](https://github.com/bitmovin/bitmovin-player-ui/pull/571/commits/15382422dfff83870af564b8e98ed49bdc356e12)